lib/db.ts

```diff
@@ -465,3 +465,53 @@ export async function deleteFile(fileId: number) {
     .where(eq(Schemas.files1.id, fileId))
     .returning();
 }
+
+
+// OTP functions ==========================================================================================
+
+export async function saveOtp(phoneNumber: string, otp: string, expiresAt: string) {
+  // Generate a salt and hash the OTP
+  const salt = genSaltSync(10);
+  const hashedOtp = hashSync(otp, salt);
+
+  return await db.insert(Schemas.otps1).values({
+    phoneNumber,
+    otp: hashedOtp,
+    expiresAt,
+  }).returning();
+}
+
+export async function deleteOtp(phoneNumber: string, otp: string) {
+  return await db.delete(Schemas.otps1)
+    .where(
+      and(
+        eq(Schemas.otps1.phoneNumber, phoneNumber),
+        eq(Schemas.otps1.otp, otp)
+      )
+    ).returning();
+}
+
+export async function verifyOtp(phoneNumber: string, otp: string): Promise<boolean> {
+  const record = await db.select()
```

Comment:
this can be done in a transaction, probably not 100% necessary, but just to make sure there are no db changes we have to consider between selecting here and the delete at the bottom of this function